src/olympia/hero/models.py

```diff
@@ -88,7 +90,32 @@ def get_choices(self, *args, **kwargs):
         return super().get_choices(*args, **kwargs)
 
 
+class PrimaryHeroImage(models.Model):
+    custom_image = models.ImageField(
+        upload_to='static/img/hero/featured',
+        blank=True)
+
+    def __str__(self):
+        return f'{self.custom_image}'
+
+    def preview_image(self):
+        return mark_safe('<img src="{}" width="150" />'.format(
+            self.custom_image.url)
+        )
+    preview_image.short_description = "Image"
+    preview_image.allow_tags = True
+
+
+class ImageChoiceField(forms.ModelChoiceField):
+    def label_from_instance(self, obj):
+        return mark_safe(
+            '<img src="{}" width="150" height="120" />'.format(
```

Comment:
same comment about defining a class in the html here - you may be able to share some css with the code above too.